src/main/scala/mesosphere/marathon/core/health/impl/MarathonHealthCheckManager.scala

```diff
@@ -51,6 +55,32 @@ class MarathonHealthCheckManager(
       ahcs(appId).values.flatten.toSet
     }
 
+  /**
+    * A common materialized merge hub that is used by all [[HealthCheckActor]]s to channel the Marathon health checks.
+    * Its main purpose is to provide a global throttling mechanism that limit a total number of concurrent Marathon
+    * health check in the system defined by [[MarathonConf.maxConcurrentMarathonHealthChecks]] parameter. After
+    * executing the health check by calling the [[HealthCheckWorker.run()]] method the result is sent back to the
+    * HealthCheckActor via the provided ActorRef.
+    */
+  val healthCheckWorkerHub: Sink[(AppDefinition, Instance, MarathonHealthCheck, ActorRef), NotUsed] =
+    MergeHub
+      .source[(AppDefinition, Instance, MarathonHealthCheck, ActorRef)](1)
+      .mapAsync(conf.maxConcurrentMarathonHealthChecks()){
+        case (app, instance, marathonHealthCheck, actorRef) =>
+          HealthCheckWorker
+            .run(app, instance, marathonHealthCheck)
+            .map(healthResult => actorRef ! healthResult)
+            .recover {
+              // Theoretically we should never get there: [[HealthCheckWorker.run]] already wraps failed health checks into
+              // successful futures with Healthy/Unhealthy message. But just in case the underlying implementation changes...
+              case NonFatal(e) =>
+                logger.warn(s"HealthCheck stream for app ${app.id} version ${app.version} " +
+                  s"and healthCheck $marathonHealthCheck failed with: ", e)
+            }
+      }
+      .to(Sink.ignore)
+      .run()
```

Comment:
I really have to read up on Akka streams now. So we materialize a hub sink that we use in another stream?

Reply:
Exactly. See its usage in the `HealthCheckActor`